### PR TITLE
fix:added fixActionPayloadForMongoQuery

### LINKED ISSUE #13228: [Bug]: Multiple Cyclic Dependency Errors
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Lately many of our dashboards have started throwing cyclic dependency errors. Looking at the logs, it looks like one of the objects in the dependency map incorrectly adds a reference to itself.

In the below case, UngroupClient.config.formData has an entry for itself. This is causing the entire application to error out and reset.

Screenshot 2022-04-22 at 7 21 16 PM

Have noticed this in at least 4 different queries, all were MongoDB queries.

While the issue is looked into, is there a way to remove the reference manually for my application?

### Steps To Reproduce

Occurs only for MongoDB queries written a few months back.

### Public Sample App

_No response_

### Version

Self Hosted - 1.6.20

### PR DESCRIPTION
**Pull request by rashmi-rai**
## Description

Added a function to fix the corrupt state of mongo query. Function adds the missing config for all the commands in mongo query.

Fixes #13228 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/mongo-query-payload 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.43 **(-0.01)** | 37.81 **(-0.05)** | 36.02 **(-0.01)** | 56.64 **(-0.01)**
 :red_circle: | app/client/src/components/formControls/utils.ts | 65.58 **(-8.66)** | 42.14 **(-18.06)** | 80.95 **(-8.52)** | 65 **(-7.73)**
 :green_circle: | app/client/src/sagas/ActionSagas.ts | 12.67 **(0.1)** | 1.34 **(-0.16)** | 8 **(0)** | 15.46 **(0.13)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

**Review by Irongade — APPROVED** [review]
Approving this as a temporary fix. We need to investgate further and come up with a scalable solution.

**Comment by Rishabh-Rathod** [comment]
> Approving this as a temporary fix. We need to investigate further and come up with a scalable solution.

@rashmi-rai  @Irongade Can we please create an issue for this task and prioritize it accordingly.

File: app/client/src/components/formControls/utils.ts [hunk 1]
```diff
@@ -278,3 +281,52 @@ export enum WhereClauseSubComponent {
 }
 
 export const allowedControlTypes = ["DROP_DOWN", "QUERY_DYNAMIC_INPUT_TEXT"];
+
+export function fixActionPayloadForMongoQuery(
```
Comment (Rishabh-Rathod) [inline app/client/src/components/formControls/utils.ts:285]: As the issue happens for all Query. @Irongade Are you aware of why we only added a fix for the mongo query here?

**Pull request merged**